### ISSUE TITLE
Session does not destroyed, when using flash

### ISSUE THREAD

**Issue by dpetrov** [PerlDancer/Dancer#1120]
Since Dancer@1.3136 the Session is cached and that might cause a problem.

I currently have an logout action like that:

```
get '/user/logout' => sub {
    session->destroy();
    flash message =>  "You have been successfully logged out.";
    redirect '/';
};
```

So session->destory() destroys the session, but since the flash writes into the session and it's cached, it creates it again and the session never got destroyed.

**Comment by yanick** [comment]
Ouch. I see the problem and have a testcase reproducing it. I'll see what can be done today.

**Comment by dpetrov** [comment]
@yanick thanks a lot. I saw the latest release a few days ago and the comment inside, so I guess I am closing the issue now ;) Should be fixed into 1.3140.
